### PR TITLE
[#1401] feat(dashboard): Support removing excluded servers

### PR DESCRIPTION
**Pull request by yl09099**
### What changes were proposed in this pull request?
1. Provide the page deletion function for blacklist pages.
2. Provide search function for blacklist page.

image
image

### Why are the changes needed?

Fix: #1401 

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

UT.

### CODE REVIEW

**Comment by yl09099** [comment]
This function needs to be reviewed after merging the previous PR.

**Comment by yl09099** [comment]
This feature is ready. PATL

File: coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java [hunk 1]
```diff
@@ -270,6 +270,24 @@ private synchronized boolean putInExcludedNodesFile(List<String> excludedNodes)
...
+    Path hadoopPath = new Path(excludedNodesPath);
+    // Obtains the existing excluded node.
+    Set<String> alreadyExistExcludedNodes =
+        parseExcludedNodesFile(hadoopFileSystem.open(hadoopPath));
+    List<String> removedAddExcludedNodes =
+        excludedNodes.stream()
+            .filter(node -> !alreadyExistExcludedNodes.contains(node))
+            .collect(Collectors.toList());
+    // Writes to the new excluded node.
+    writeExcludedNodes2File(removedAddExcludedNodes);
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java:287]:
```suggestion
writeExcludedNodes2File(alreadyExistExcludedNodes.removeAll(excludedNodes));
```

File: coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java [hunk 2]
```diff
@@ -270,6 +270,24 @@ private synchronized boolean putInExcludedNodesFile(List<String> excludedNodes)
     return true;
   }
 
+  private synchronized boolean removeExcludedNodesFile(List<String> excludedNodes)
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java:273]:
```suggestion
  private synchronized boolean removeExcludedNodesFromFile(List<String> excludedNodes)
```

File: dashboard/src/main/webapp/src/pages/serverstatus/ExcludeNodeList.vue [hunk 3]
```diff
@@ -136,14 +135,115 @@ export default {
...
+        })
+        .catch(() => {
+          ElMessage({
+            type: 'info',
+            message: 'Delete canceled'
+          })
+        })
+    }
+
+    async function deleteShuffleExcludeNodesPage(excludeNodes) {
```
Comment (xianjingfeng) [inline dashboard/src/main/webapp/src/pages/serverstatus/ExcludeNodeList.vue:201]:
```suggestion
    async function deleteShuffleExcludedNodes(excludeNodes) {
```

File: dashboard/src/main/webapp/src/pages/serverstatus/ExcludeNodeList.vue [hunk 4]
```diff
@@ -17,19 +17,32 @@
 
 <template>
   <div>
-    <div style="text-align: right">
-      <el-button type="primary" @click="dialogFormVisible = true"> Add Node </el-button>
+    <div class="button-wrapper">
+      <el-button type="success" @click="dialogFormVisible = true"> Add Node </el-button>
+      <el-button type="danger" @click="handlerDeleteNode">Delete({{ selectItemNum }})</el-button>
```
Comment (xianjingfeng) [inline dashboard/src/main/webapp/src/pages/serverstatus/ExcludeNodeList.vue:22]:
```suggestion
      <el-button type="danger" @click="handleDeleteNode">Delete({{ selectItemNum }})</el-button>
```

File: coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java [hunk 5]
```diff
@@ -383,6 +398,18 @@ public boolean addExcludedNodes(List<String> excludedNodeIds) {
     }
   }
 
+  @Override
+  public boolean removeExcludedNodesFromFile(List<String> excludedNodeIds) {
+    try {
+      boolean successFlag = removeExcludedNodesFile(excludedNodeIds);
+      excludedNodes.removeAll(excludedNodeIds);
+      return successFlag;
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java:406]:
```suggestion
      if(removeExcludedNodesFile(excludedNodeIds)) {
        excludedNodes.removeAll(excludedNodeIds);
        return true;
      }
      return false;
```

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Pull request merged**